galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/RestoreCPS.java

```diff
@@ -0,0 +1,294 @@
+    
+    private Map<String, IConfigurationPropertyStoreService>     namespaceCPS 
+                            = new HashMap<String ,IConfigurationPropertyStoreService>();
+    
+    /**
+     * <p>Restores configuration properties from the specified file to the Configuration Property Store (CPS)</p>
+     * 
+     * @param bootstrapProperties
+     * @param overrideProperties
+     * @param filePath
+     * @return
+     * @throws FrameworkException
+     */
+    public void restore(Properties bootstrapProperties, Properties overrideProperties, String filePath) throws FrameworkException {
+        logger.info("Initialising CPS Restore Service");
+        
+        FrameworkInitialisation frameworkInitialisation = null;
+        try {
+            frameworkInitialisation = new FrameworkInitialisation(bootstrapProperties, overrideProperties);
+        } catch (Exception e) {
+            throw new FrameworkException("Unable to initialise the Framework Service", e);
+        }
+        
+        framework = frameworkInitialisation.getFramework();
+        
+        Properties properties = getProperties(filePath);
+        if (properties != null) {
+            restoreProperties(properties);
+        } else {
+            logger.info("No properties found to restore.");
+        }
+    }
+    
+    /**
+     * <p>Fetches configuration properties from a specified file</p>
+     * 
+     * @param filePath
+     * @return properties
+     */
+    private Properties getProperties(String filePath) {
+        
+        Properties properties = new Properties();
+        
+        path = Paths.get(filePath);
+        InputStream inputStream = null;
+        try {
+            inputStream = Files.newInputStream(path);
+        } catch (IOException e) {
+            logger.error("Error Creating InputStream Using File: " + path.toUri().toString(), e);
+        }
+        
+        if (inputStream != null) {
+            try {
+                properties.load(inputStream);
+            } catch (IOException e) {
+                logger.error("Error Loading Properties From InputStream", e);
+            }
+        }
+        
+        return properties;
+
+    }
+    
+    /**
+     * <p>Iterates through the properties in the supplied object, restoring them one-by-one to the Configuration Property Store </p>
+     * 
+     * @param properties (instance of object: Properties)
+     * @return 
+     */
+    private void restoreProperties(Properties properties) throws ConfigurationPropertyStoreException {
+        
+        logger.info(String.format("|%-12s|%-17s|%-32s|%-17s|%-17s|", getDashes(12), getDashes(17), getDashes(32), getDashes(17), getDashes(17)));
+        logger.info(String.format("| %-10s | %-15s | %-30s | %-15s | %-15s |", "STATUS", "NAMESPACE", "PROPERTY", "OLD VALUE", "NEW VALUE"));
+        logger.info(String.format("|%-12s|%-17s|%-32s|%-17s|%-17s|", getDashes(12), getDashes(17), getDashes(32), getDashes(17), getDashes(17)));
```

Comment:
> @Michael-Baylis and I had a discussion on this topic and decided that it would be best to drop the table feature and simply log the output line-by-line.